fardel/apps/ecommerce/checkout/views.py

```diff
@@ -159,3 +160,76 @@ def delete(self):
         return {
             "message":"successfuly cleared the shopping cart."
         }
+
+
+@rest_resource
+class PaymentApi(Resource):
+    """
+    :URL: ``/api/ecommerce/checkout/payment/``
+    """
+    endpoints = ['/checkout/payment/', '/checkout/payment/payment_id']
+
+    @jwt_required
+    def get(self, payment_id):
+
+        """ to get payemnt of a current_user with this cart_id """
+
+        payemnt = Payment.query.filter_by(id=payment_id).first()
+        if payment:
+            if current_user and payment.user_id != current_user.id:
+                return {"payment": None}
+            return {"payment": payment.dict()}
+
+        return {"payment": None}
```

Comment:
in get method:
if payment_id is available:
    check verification of payment ( if the payment is waiting )
else:
    paginate over user's payment